### PR TITLE
feat/studio/usage: Additional project usage charts/tables

### LINKED ISSUE #4796: Add chart and table for some more REST analytics on project overview
Show API status codes chart on project overview. Should give us something like:

Screen Shot 2022-01-04 at 11 20 59 AM

Logflare prod endpoint ID: 66

Show API response times, etc table under status codes chart. Should give us something like:

Screen Shot 2022-01-04 at 11 20 39 AM

Logflare prod endpoint ID: 67

Stories:
- [ ] User should be able to click on a row in the table and get taken to the API log search with the appropriate SQL query 
- [ ] User should be able to click the chart and get taken to the API log search page with the appropriate SQL query
- [ ] Hour / day / week picker on the project overview should affect these data as well

### PR DESCRIPTION
**Pull request by Ziinc**
Closes #4796 , adds an additional chart and table to the project usage stats page.

- [x] additional data fetching
- [x] StackedAreaChart implementation
- [ ] tests

### CODE REVIEW

**Comment by Ziinc** [comment]
Latest SS:

image

**Comment by Ziinc** [comment]
Latest demo:

https://user-images.githubusercontent.com/22714384/154316371-8e4b4466-19d4-4295-a547-e80088ce4227.mov

**Comment by Ziinc** [comment]
Pre-merge checklist:
- [ ] merge in endpoint proxying https://github.com/supabase/infrastructure/pull/4434
- [ ] add in feature flag 
- [ ] update prod endpoints (chart/codes) to use new sql query

**Comment by joshenlim** [comment]
Updated the status code chart colors but im not convinced that the colors i chose are correct

image

Probably need @MildTomato 's input on this

Also could we hold off this PR until the UI library updates are in too? Foresee some conflicts

**Review by joshenlim — APPROVED** [review]
i think let's just get this in first! PR lgtm :) 

needs relooking at the colours and placing eventually but I suppose we can figure out these later too

**Comment by Ziinc** [comment]
OK cool, colors look great. I'll get this merged in today.

I think would be good to have a color palette in the ui library, so that we
can use exported constants instead of hardcoding the hexes each time.

**Comment by MildTomato** [comment]
> OK cool, colors look great. I'll get this merged in today.
> 
> I think would be good to have a color palette in the ui library, so that we
> can use exported constants instead of hardcoding the hexes each time.

I've updated the UI library (alpha release) to use CSS variables now.

So you can use colors like `--colors-scale-900` rather than hardcoding hex colors in :D

File: studio/components/ui/Charts/Charts.constants.ts
Comment (MildTomato) [inline studio/components/ui/Charts/Charts.constants.ts:13]: @Ziinc @joshenlim 

nice, we should be able to use CSS variables in here instead 🥳

Comment (joshenlim) [reply studio/components/ui/Charts/Charts.constants.ts:13]: sounds good! yup we can do this once the [UI library PR](https://github.com/supabase/supabase/pull/5463) is in
just curious though how will this look like in JS?

is it something like `import { colors_scale } from @supabase/ui`?

**Pull request merged**
